STAR.Web/Controllers/SkillController.cs

```diff
@@ -19,20 +19,43 @@ public SkillController(DbContext context)
         }
 
         [HttpPost]
-        public ActionResult Skills(SkillModel model)
+        public ActionResult Details(SkillModel model)
         {
+            if (context.Skills.Any(x => x.Name == model.SkillName))
+            {
+                return View("SkillExists");
```

Comment:
Instead of throwing the user to another view, invalidate the Model, and give an error message that will lead the user to correct the problem.